finish/src/main/java/io/openliberty/guides/rest/PropertiesResource.java

```diff
@@ -28,17 +27,8 @@
     // tag::getProperties[]
     @GET
     @Produces(MediaType.APPLICATION_JSON)
```

Comment:
Noting that `java.util.Properties` is, AFAIK, only mapped by Yasson... not by the JSONB spec.  Not sure if we try to distinguish those types of usages.. guessing not since I don't recall seeing anything like that in any of the guide variants I've looked at.

Reply:
Then, Yasson is the reference implementation for JSON-B.  Interesting.  We do call out details/subtleties relevant to developers where appropriate.  If this is not specification, even if Yasson is the reference implementation, I am fine to note it.  Makes sense to do so too.

Reply:
`java.util.Properties` implements `java.util.Map` and is therefore covered by JSON-B spec. Yasson has no special handling for `java.util.Properties` and any other JSON-B spec compliant impl should work too

Reply:
Hmm.. I must have misinterpreted my experience running some of these samples in TomEE and failing.  Maybe it's a bug in Johnzon or even something else in the JAX-RS layers.   I'd have to look into it more and don't want to hold anything up for now then.

Reply:
@scottkurz a while back, we chatted about a Johnson-specific issue with mapping Properties:
```
Caused by: javax.json.bind.JsonbException: Can't map class java.util.Properties
       at org.apache.johnzon.jsonb.JohnzonJsonb.fromJson(JohnzonJsonb.java:200)
       at org.apache.johnzon.jaxrs.jsonb.jaxrs.JsonbJaxrsProvider.readFrom(JsonbJaxrsProvider.java:182)
       at org.apache.cxf.jaxrs.utils.JAXRSUtils.readFromMessageBodyReader(JAXRSUtils.java:1385)
       at org.apache.cxf.jaxrs.impl.ResponseImpl.doReadEntity(ResponseImpl.java:370)
       ... 50 more
Caused by: org.apache.johnzon.mapper.MapperException: Can't map class java.util.Properties
       at org.apache.johnzon.mapper.MappingParserImpl.buildObject(MappingParserImpl.java:277)
       at org.apache.johnzon.mapper.MappingParserImpl.readObject(MappingParserImpl.java:141)
       at org.apache.johnzon.mapper.MappingParserImpl.readObject(MappingParserImpl.java:133)
       at org.apache.johnzon.mapper.MappingParserImpl.readObject(MappingParserImpl.java:125)
       at org.apache.johnzon.mapper.Mapper.mapObject(Mapper.java:254)
       at org.apache.johnzon.mapper.Mapper.readObject(Mapper.java:196)
       at org.apache.johnzon.jsonb.JohnzonJsonb.fromJson(JohnzonJsonb.java:194)
       ... 53 more
```

I think it is a bug with Johnzon - I'm not sure, but I think the problem is in this method ( https://github.com/apache/johnzon/blob/2807054bcc9ae43ebdfe73b0846d6b9515eef21e/johnzon-mapper/src/main/java/org/apache/johnzon/mapper/MappingParserImpl.java#L268 ) - I think it should be checking whether type can be assigned to `Map` rather than checking that it _is_ a `Map`.

Reply:
Learning a lot here..thx for explaining.